java/bulk-import/bulk-import-starter/src/main/java/sleeper/bulkimport/starter/executor/ExecutorFactory.java

```diff
@@ -38,22 +42,29 @@ public class ExecutorFactory {
     private final InstanceProperties instanceProperties;
     private final TablePropertiesProvider tablePropertiesProvider;
     private final StateStoreProvider stateStoreProvider;
+    private final IngestJobStatusStore ingestJobStatusStore;
     private final AmazonS3 s3Client;
     private final AmazonElasticMapReduce emrClient;
     private final AWSStepFunctions stepFunctionsClient;
     private final String bulkImportPlatform;
+    private final String taskId;
+    private final Supplier<Instant> validationTimeSupplier;
 
     public ExecutorFactory(AmazonS3 s3Client,
                            AmazonElasticMapReduce emrClient,
                            AWSStepFunctions stepFunctionsClient,
-                           AmazonDynamoDB dynamoDB) throws IOException {
-        this(s3Client, emrClient, stepFunctionsClient, dynamoDB, System::getenv);
+                           AmazonDynamoDB dynamoDB,
+                           String taskId,
+                           Supplier<Instant> validationTimeSupplier) throws IOException {
```

Comment:
Can we keep the code for defaulting taskId and validationTimeSupplier inside this class, to match getEnvironmentVariable, rather than doing it in BulkImportStarter?